src/qml/bitcoin.h

```diff
@@ -5,6 +5,77 @@
 #ifndef BITCOIN_QML_BITCOIN_H
 #define BITCOIN_QML_BITCOIN_H
 
+#include <common/args.h>
+#include <interfaces/init.h>
+
+#include <interfaces/node.h>
+#include <qt/initexecutor.h>
+
+#include <assert.h>
+#include <memory>
+#include <optional>
+
+#include <boost/signals2/connection.hpp>
+
+#include <QObject>
+#include <QGuiApplication>
+#include <QQmlApplicationEngine>
+#include <QQuickWindow>
+#include <QThread>
+
+class NodeModel;
+class ChainModel;
+class OptionsQmlModel;
+class NetworkTrafficTower;
+class PeerTableModel;
+class PeerListSortProxy;
+class OnboardingModel;
+
+class BitcoinQmlApplication: public QGuiApplication
```

Comment:
the logic behind having a separate class came from qt/bitcoin.h and cpp where the class is set to accomodate member variables... Do you think this can be achieve with just the anymous "namespace" within qml/bitcoin.cpp?

Reply:
Ok got the custom datadir to work with an anonymous namespace in separate branch... Will update and push the code in a new PR once the UI ones have been merged